### ISSUE TITLE
CheckBox should provide a mode to control rendering of nulls as indeterminate vs. unchecked

### ISSUE THREAD

**Issue by febbraiod** [exhi/hoist-react#841]
We have a client app that does not always return values for controls that should be treated as 'false'. Would be great to not have to specify which fields need to be converted. If our checkbox could be told to treat nulls and undefined as false, that would eliminate the need to maintain this kind of fiddly code in the client app.

**Comment by amcclain** [comment]
My immediate reaction to this ticket was that the checkbox was acting as a bit of a canary, indicating that the underlying app code really should/could coerce nulls to false if a T/F model was really what was required. 

@lbwexler made the good point, however, that we want to be careful with such routines that would modify data on load to client / population into a control, when otherwise the user might not have interacted with or otherwise meant to change the data value.

And I do agree that the indeterminate [-] checkbox is plain weird and not likely to be understood by many users.  

So in short - I support this change, and think it should be the default behavior of our checkbox to render null as unchecked.
